psse/psse-converter/src/main/java/com/powsybl/psse/converter/PsseImporter.java

```diff
@@ -182,30 +211,99 @@ private static void createLoad(PsseLoad psseLoad, ContainersMapping containerMap
+        }
+    }
+
+    private void createSwitchedShuntBlocMap(PsseRawModel psseModel, HashMap<PsseSwitchedShunt, ShuntBlockTab > stoBlockiTab) {
+
+        /* Creates a map between the PSSE switched shunt and the blocks info of this shunt
+        A switched shunt may contain up to 8 blocks and each block may contain up to 9 steps of the same value (in MVAR)
+        A block may be capacitive or inductive */
+        for (PsseSwitchedShunt psseSwShunt : psseModel.getSwitchedShunts()) {
+
+            ShuntBlockTab sbt = new ShuntBlockTab();
+
+            int[] ni = new int[8];
+            ni[0] = psseSwShunt.getN1();
+            ni[1] = psseSwShunt.getN2();
+            ni[2] = psseSwShunt.getN3();
+            ni[3] = psseSwShunt.getN4();
+            ni[4] = psseSwShunt.getN5();
+            ni[5] = psseSwShunt.getN6();
+            ni[6] = psseSwShunt.getN7();
+            ni[7] = psseSwShunt.getN8();
+
+            double[] bi = new double[8];
+            bi[0] = psseSwShunt.getB1();
+            bi[1] = psseSwShunt.getB2();
+            bi[2] = psseSwShunt.getB3();
+            bi[3] = psseSwShunt.getB4();
+            bi[4] = psseSwShunt.getB5();
+            bi[5] = psseSwShunt.getB6();
+            bi[6] = psseSwShunt.getB7();
+            bi[7] = psseSwShunt.getB8();
+
+            int i = 0;
+            while (i <= 7 && ni[i] > 0) {
+                sbt.add(i + 1, ni[i], bi[i]);
+                i++;
+            }
+
+            stoBlockiTab.put(psseSwShunt, sbt);
         }
+    }
 
-        if (psseShunt.getGl() != 0) {
-            LOGGER.warn("Shunt Gl not supported ({})", psseShunt.getI());
+    private static void createSwitchedShunt(PsseSwitchedShunt psseSwShunt, PerUnitContext perUnitContext, ContainersMapping containerMapping, Network network, HashMap<PsseSwitchedShunt, ShuntBlockTab >  stoBlockiTab) {
+        String busId = getBusId(psseSwShunt.getI());
+        VoltageLevel voltageLevel = network.getVoltageLevel(containerMapping.getVoltageLevelId(psseSwShunt.getI()));
+        ShuntBlockTab sbl = stoBlockiTab.get(psseSwShunt);
+
+        for (int i = 1; i <= sbl.getSize(); i++) {
+            if (psseSwShunt.getBinit() != 0) { //TODO : improve it to make it robust to all configurations
+                ShuntCompensatorAdder adder = voltageLevel.newShuntCompensator()
+                        .setId(busId + "-SwSH-B" + i)
+                        .setConnectableBus(busId)
+                        .setSectionCount(1);
+                adder.newLinearModel() //TODO: use Binit to initiate Bi, for now we use Binit to obtain de same load-flow results
+                        .setBPerSection(psseSwShunt.getBinit())//TODO: take into account BINIT to define the number of switched steps in the case BINIT is different from the max switched steps
+                        .setMaximumSectionCount(1)
+                        //.setCurrentSectionCount(sbl.getNi(i))
+                        //.setMaximumSectionCount(sbl.getNi(i))
```

Comment:
Is it ok to comment these lines?

Reply:
OK removed, still to do the right initialization of B